### PR TITLE
FAI-241: Model definition endpoint - UI changes

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-241

This PR adds a Model's `namespace` to the information stored for executions (it is already in `TraceEvent` but was not being persisted). We can then lookup a Model's definition from the `executionId` (that internally looks up the `modelName` and `modelNamespace` associated with the `executionId` and subsequently model using the `modelId`). REST endpoints changed accordingly. Model is now looked up with `v1/executions/{executiondId}/model` (let me know if you want versioning of the REST endpoint added now I've added a new path, i.e. `v2/executions/{executionId}/model`.. but for now this is all internal so didn't think we needed to worry too much about versioning the endpoint atm).

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by manstis** [comment]
`kogito-runtimes` has test failures:
```
[ERROR] Errors: 
[ERROR]   AdHocFragmentsTest.testUserTaskProcess » Runtime io.quarkus.bootstrap.Bootstra...
[ERROR] Tests run: 27, Failures: 0, Errors: 1, Skipped: 26
```
@ricardozanini ^^ FYI. Something you've changed?

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by ricardozanini** [comment]
> `kogito-runtimes` has test failures:
> 
> ```
> [ERROR] Errors: 
> [ERROR]   AdHocFragmentsTest.testUserTaskProcess » Runtime io.quarkus.bootstrap.Bootstra...
> [ERROR] Tests run: 27, Failures: 0, Errors: 1, Skipped: 26
> ```
> 
> @ricardozanini ^^ FYI. Something you've changed?

Hi! Not at all!

My PR stills opened.

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/ExecutionsApiV1.java [hunk 1]
```diff
@@ -142,4 +146,52 @@ private OffsetDateTime parseParameterDate(String date) {
...
+    private Response handleModelRequest(String executionId) {
+        return retrieveModel(executionId)
+                .map(definition -> Response.ok(definition).build())
+                .orElseGet(() -> Response.status(Response.Status.BAD_REQUEST.getStatusCode()).build());
+    }
+
+    private Optional<String> retrieveModel(String executionId) {
+        try {
+            Optional<Decision> decision = retrieveDecision(executionId);
+            return decision.map(d -> executionService.getModelById(ModelIdCreator.makeIdentifier(null, null, null, d.getExecutedModelName(), d.getExecutedModelNamespace())));
```
Comment (r00ta) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/ExecutionsApiV1.java:185]: add TODO comment to replace `null` `null`?

**Review by r00ta — APPROVED** [review]

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/ExecutionsApiV1.java (hunk 1 above)
Comment (manstis) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/ExecutionsApiV1.java:185]: Done (in new commit).

**Comment by manstis** [comment]
@jiripetrlik This has been rebased following @r00ta changes and is ready for your review. Thank-you.

**Review by jiripetrlik — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

**Review by evacchi — APPROVED** [review]
LGTM

**Pull request merged**